src/core/ControllerConnection.cpp

```diff
@@ -211,10 +211,12 @@ void ControllerConnection::loadSettings( const QDomElement & _this )
 			m_controllerId = -1;
 		}
 
-		if (!Engine::getSong()->isLoadingProject() && m_controllerId != -1)
+		if ( !Engine::getSong()->isLoadingProject()
+			&& m_controllerId != -1
+			&& m_controllerId < Engine::getSong()->controllers().size() )
```

Comment:
I think you can move this check to the index validation a few lines above if you want.

Reply:
Moved. It does make more sense to have all the range-checks in one if() instead of spread between the two.

Reply:
This check should have been there because the two `if` statements do different things.
Reported by @zonkmachine:

> With commit 91f9f1a from #4939 I get an issue with loading a project with a controller.
> To replicate:
>  1. In a new default project, connect an LFO to the 3osc pan knob in the song editor.
>  2. Save and reload.
> The project loads but the knob isn't moving. Command line output reads controller index invalid.

This was my mistake.

Reply:
Fixed via 934ea6d0866818833caaacbe16c816b9a94c9b9a.

Reply:
@PhysSong Fix confirmed, thanks!